server/src/main/java/org/elasticsearch/common/util/BigIntArray.java

```diff
@@ -54,11 +54,17 @@ public void writeTo(StreamOutput out) throws IOException {
         }
         int intSize = (int) size;
         out.writeVInt(intSize * 4);
+        int lastPageEnd = intSize % INT_PAGE_SIZE;
```

Comment:
I think you could fix this with a one liner like `int lastPageEnd = (intSize + 1) % INT_PAGE_SIZE - 1;` but I think that's harder to read that this.

Reply:
Nit: on the line above, we should use `Integer.BYTES`, not a hard-coded 4